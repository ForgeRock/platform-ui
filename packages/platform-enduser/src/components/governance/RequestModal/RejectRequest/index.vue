<!-- Copyright (c) 2023 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <div class="p-25">
    <div class="p-25">
      {{ subtitle() }}
    </div>
    <FrField
      @input="$emit('request-comment', $event);"
      class="mb-4"
      type="textarea"
      validation="required"
      name="rejectComment"
      :label="$t('governance.requestModal.reject.justification')"
      :description="$t('governance.requestModal.reject.footer')" />
  </div>
</template>

<script>

import FrField from '@forgerock/platform-shared/src/components/Field';

export default {
  name: 'RejectRequest',
  components: { FrField },
  methods: {
    subtitle() {
      return this.$t('governance.requestModal.subtitle', { modal: this.$t('common.reject').toLowerCase() });
    },
  },
};
</script>
<style scoped>
  .p-25 {
    padding-bottom: 25px;
  }
</style>
