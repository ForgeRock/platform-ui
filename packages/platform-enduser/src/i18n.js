/**
 * Copyright (c) 2020-2024 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { createI18n } from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

export default createI18n({
  messages: loadLocaleMessages(),
  silentFallbackWarn: true,
  silentTranslationWarn: true,
  dateTimeFormats,
});
