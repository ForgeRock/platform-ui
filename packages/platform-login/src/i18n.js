/**
 * Copyright (c) 2020-2021 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import Vue from 'vue';
import VueI18n from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

Vue.mixin({
  methods: {
    translationExists(path) {
      return this.$t(path) !== path;
    },
  },
});

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
