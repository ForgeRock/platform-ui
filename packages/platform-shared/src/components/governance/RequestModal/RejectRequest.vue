<!-- Copyright (c) 2023-2024 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <div class="pb-25px">
    <div class="pb-25px">
      {{ $t('governance.requestModal.subtitle', { modal: $t('common.reject').toLowerCase() }) }}
    </div>
    <FrField
      @input="$emit('request-comment', $event);"
      class="mb-4"
      type="textarea"
      :validation="{ required: requireJustification }"
      name="rejectComment"
      :label="$t('governance.requestModal.justification')"
      :description="$t('governance.requestModal.reject.justificationDescription')" />
  </div>
</template>

<script setup>
/**
 * Displays the modal with a subtitle and justification field to reject a request.
 * @component RejectRequest
 */
import FrField from '@forgerock/platform-shared/src/components/Field';

defineProps({
  requireJustification: {
    default: false,
    type: Boolean,
  },
});

defineEmits(['request-comment']);
</script>

<style lang="scss" scoped>
.pb-25px {
  padding-bottom: 25px;
}
</style>
