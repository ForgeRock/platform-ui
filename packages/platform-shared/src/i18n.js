/**
 * Copyright 2019 ForgeRock AS. All Rights Reserved
 *
 * Use of this code requires a commercial software license with ForgeRock AS.
 * or with one of its affiliates. All use shall be exclusively subject
 * to such license between the licensee and ForgeRock AS.
 */
import Vue from 'vue';
import VueI18n from 'vue-i18n';
import dateTimeFormats from './dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      messages[locale] = locales(key);
    }
  });

  return messages;
}

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
