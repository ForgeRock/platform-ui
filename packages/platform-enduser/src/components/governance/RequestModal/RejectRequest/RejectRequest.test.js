/**
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { mount } from '@vue/test-utils';
import RejectRequest from './index';
import i18n from '@/i18n';

describe('RejectRequest', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(RejectRequest, {
      mocks: {
        $t: (text, prop) => i18n.t(text, prop),
      },
    });
  });

  it('Subtitle should contain reject', async () => {
    expect(wrapper.vm.subtitle()).toEqual('Are you sure you want to reject this request?');
  });
});
