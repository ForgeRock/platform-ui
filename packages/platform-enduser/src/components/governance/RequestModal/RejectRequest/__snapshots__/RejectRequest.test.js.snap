// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RejectRequest Subtitle should contain reject: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) vm.$listeners has been removed. In Vue 3, parent v-on listeners are included in vm.$attrs and it is no longer necessary to separately use v-on="$listeners" if you are already using v-bind="$attrs". (Note: the Vue 3 behavior only applies if this compat config is disabled)
  Details: https://v3-migration.vuejs.org/breaking-changes/listeners-removed.html",
    "
",
    " at <FrField",
    "onInput=fn",
    "class="mb-4"",
    "type="textarea"",
    " ...",
    ">",
    "
",
    " at <RejectRequest",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation WATCH_ARRAY) (2)",
    "
",
    " at <TextArea",
    "onInput=",
    [
      [Function],
      [Function],
    ],
    "type="textarea"",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <FrField",
    "onInput=fn",
    "class="mb-4"",
    "type="textarea"",
    " ...",
    ">",
    "
",
    " at <RejectRequest",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;
